parquet-variant/src/to_json.rs

```diff
@@ -0,0 +1,1272 @@
+            } else {
+                let divisor = 10_i64.pow(*scale as u32);
+                let quotient = integer / divisor;
+                let remainder = (integer % divisor).abs();
+                let formatted_remainder = format!("{:0width$}", remainder, width = *scale as usize);
+                let trimmed_remainder = formatted_remainder.trim_end_matches('0');
+
+                let decimal_str = if trimmed_remainder.is_empty() {
+                    quotient.to_string()
+                } else {
+                    format!("{}.{}", quotient, trimmed_remainder)
+                };
+
+                // Parse as serde_json::Number to preserve precision
+                decimal_str
+                    .parse::<serde_json::Number>()
+                    .map(Value::Number)
+                    .map_err(|e| {
+                        ArrowError::InvalidArgumentError(format!("Invalid decimal string: {}", e))
+                    })
+            }
+        }
+        Variant::Decimal16(VariantDecimal16 { integer, scale }) => {
+            // Use integer arithmetic to avoid f64 precision loss
+            if *scale == 0 {
+                Ok(Value::Number((*integer as i64).into())) // Convert to i64 for JSON compatibility
+            } else {
+                let divisor = 10_i128.pow(*scale as u32);
+                let quotient = integer / divisor;
+                let remainder = (integer % divisor).abs();
+                let formatted_remainder = format!("{:0width$}", remainder, width = *scale as usize);
+                let trimmed_remainder = formatted_remainder.trim_end_matches('0');
+
+                let decimal_str = if trimmed_remainder.is_empty() {
+                    quotient.to_string()
+                } else {
+                    format!("{}.{}", quotient, trimmed_remainder)
+                };
+
+                // Parse as serde_json::Number to preserve precision
+                decimal_str
+                    .parse::<serde_json::Number>()
+                    .map(Value::Number)
+                    .map_err(|e| {
+                        ArrowError::InvalidArgumentError(format!("Invalid decimal string: {}", e))
+                    })
+            }
+        }
+        Variant::Date(date) => Ok(Value::String(format_date_string(date))),
+        Variant::TimestampMicros(ts) => Ok(Value::String(ts.to_rfc3339())),
+        Variant::TimestampNtzMicros(ts) => Ok(Value::String(format_timestamp_ntz_string(ts))),
+        Variant::Binary(bytes) => Ok(Value::String(format_binary_base64(bytes))),
+        Variant::String(s) => Ok(Value::String(s.to_string())),
+        Variant::ShortString(s) => Ok(Value::String(s.to_string())),
+        Variant::Object(obj) => {
+            let mut map = serde_json::Map::new();
+
+            for (key, value) in obj.iter() {
+                let json_value = variant_to_json_value(&value)?;
+                map.insert(key.to_string(), json_value);
+            }
+
```

Comment:
`serde_json::Map` implements [FromIterator](https://docs.rs/serde_json/latest/serde_json/struct.Map.html#impl-FromIterator%3C(String,+Value)%3E-for-Map%3CString,+Value%3E):
```suggestion
            let map = obj
                .iter
                .map(|(k, v)| Ok((k.to_string(), variant_to_json_value(&v)?)))
                .collect::<Result<_, _>>()?;
```